### DISCUSSION TITLE
Mining documentation

### DISCUSSION THREAD

**Discussion by tbremard** [dogecoin/dogecoin#2818] [Q&A] [answered]
Hello,

I ve started today to look for information about mining on DogeCoin, and arrived to the official website of dogecoin.com
I ve seen documentation on wallet, and installed 'wallet core', which is currently syncrhronising , but the menus do not propose any 'mining' functionnality.
However I am lost to find any official documentation of how to mine DogeCoin. Is it a strategy to hide it ? Are you aware that lot of youtube videos ask their viewers to download trojan app like unmineable.com?
On the github repo of DogeCoin I have been looking something like 'DogeCoin miner' but didn't find either any relevant answer (only a miner.cpp), same issue with the FAQ of DogeCoin.com, and in he discussion history (here)
I would like to be able to read somewhere if DogeCoin mining is private/ protected, or made public. If it is public to understand the guidelines.

Among other things I would like to know if on a windows desktop it is possible.
Regards.

**Comment by DirtMcGirt27** [comment]
This sounds like it may be some confusion that has arisen from the PoS (Proof of Stake, not to be confused with Point of Sale) 'proposal' brought up recently.  The core wallet will provide you with all of the functions needed to run a node, as well as manage all of your coins.  This also helps to support the network, as you are hosting a copy of the blockchain.

Mining itself is PoW (Proof of Work) and can only be performed using a Scrypt ASIC miner with any efficiency (manual hashing is a thing, though has always been a novelty).  The currently listed info is dated though a solid starting point for understanding the functionality.  

It is no longer feasible to do on a standalone Windows desktop without the ASIC mentioned

**Comment by patricklodder** [comment, marked as answer]
Hi!

> Is it a strategy to hide it ?

Not to my knowledge. There have been a couple of guides posted on reddit.com but those have to my knowledge not been peer-reviewed, so I cannot vouch for their correctness. So far, there have been no submissions to this repository of well written, comprehensive mining documentation.

The main problem is that to effectively mine, you do not only need Dogecoin Core software. And it is hard to vouch for 3rd party software inside this repository. I heard there is a website update for dogecoin.com coming, maybe this is covered there, and if it is not, maybe we can add and maintain it there, or provide links to vetted information.

On this repository, we will probably have to restrict it to more technical information, but I also think that this is a worthy goal to have.

> I would like to be able to read somewhere if DogeCoin mining is private/ protected, or made public. If it is public to understand the guidelines.

Just to answer this right now: Dogecoin mining is permissionless, just like Bitcoin mining. Anyone can submit a valid block. Because of the widespread use of specialized Scrypt hashing equipment, the difficulty of doing so is of course quite high and it's unlikely to perform CPU or GPU mining profitably and one may need to make an investment in equipment. I've read a couple of stories about how people start with cheaper, older miner equipment, turn that into a profit and buy bigger, more modern equipment from that.

**Comment by tbremard** [comment]
So if I understand correctly:
There is no Mining software available for computer with standard OS (Linux, Mac, Windows)
It is only possible to mine DogeCoin on dedicated hardware based on ASIC.
Is there official guidelines of DogeCoin to know how to configure Asic ? Which software should be used on Asic ? How to choose Asic to buy ? which chips ? How to configure it  to connect it on Mining network ?

I am missing the technical explanation why Computer software could not emulate Asic workload algorithm, even if I am aware that performance should be less.

**Reply by patricklodder** [reply]
> There is no Mining software available for computer with standard OS (Linux, Mac, Windows)

Of course not. It is available and built-in to this software. try `dogecoin-cli help generate` (or simply `help generate` on the rpc console of the QT wallet) to get help on using the built-in miner. It's just that the odds that you will mine an actual block on the Dogecoin main network are extremely low.

To see how mining works, you can try it on testnet (starting your dogecoind or dogecoin-qt with the `-testnet` command line argument) and you will be able to mine blocks with a CPU there, as there is no incentive to run optimized hardware for coins that aren't worth anything.

**Reply by tbremard** [reply]
Hello Patrick,
thank you for your answer, I am happy to learn that the wallet core have tools to mine.
Indeed in Help menu/Debug Window/Console tab the command:
help generate
gives some hint, but if I try to generate 1 as an experiment and without time limit i obtain an empty result as shown below [] 

![image](https://user-images.githubusercontent.com/36224228/151123195-adb94df0-4159-4c9d-b5a4-2c038ae12790.png)

I do not understand why you suggest to use the test network with the -testnet.
I would like to be able to try to generate real dogecoins even if low performance on real blockchain and put some reward in my dogecoin wallet, if it is possible.

**Reply by patricklodder** [reply]
> without time limit i obtain an empty result as shown below []
> [..]
> I would like to be able to try to generate real dogecoins even if low performance on real blockchain and put some reward in my dogecoin wallet, if it is possible.

Based on the timestamps in your screenshot, your CPU performed 1 million hashes in 22 seconds. That means your performance is about 45,454 hashes per second. The [total estimated hashes](https://www.coinwarz.com/mining/dogecoin/hashrate-chart) per second of the network is `396,678,884,491,264` at the time I write this.

This means that your share of the hash rate is `45,454 / 396,678,884,491,264  = 0.000000000114586`. With a block time of 1 minute, and given that the hash rate won't grow, you would statistically be able to find one block every 872,706,962.5 minutes. That is once in 606,046 days, or 1660 years. I wouldn't recommend you to run a pc for 1660 years to get 10,000 DOGE.

This is why we say that to **effectively** mine (get any reward at all within a reasonable timeframe) the odds are much better with specialized hardware (that in turn comes with specialized software) and work in a pool if you're just starting out, so that you get more frequent income. There is a comparison of scrypt hardware and pools available on [minerstat.com](https://minerstat.com/algorithm/scrypt/profitability).

The reason why I suggested to try it on testnet is because it won't time out there, which is purely a learning exercise.

**Reply by tbremard** [reply]
Thank you Patrick for these interesting explanations.

**Reply by tealethetoolman** [reply]
Just a reminder, yeah the odds are against you as a solo miner, but the idea of immutable money should be incentive enough for you to put your extra resources into the cause despite the major pools. We’re lucky they’re not hoarding the code checkouts like the major btc pools do.

**Reply by tbremard** [reply]
Thank you tealethetoolman , I am not sure to understand you correctly:

> put your extra resources into the cause despite the major pools

If you suggest to join a mining pool, could you explain how ? The topic of the discussion is to explain the different methods , platforms to mine DogeCoin, because no official documentation is available on the very website neither on this git hub repo.

**Reply by tealethetoolman** [reply]
being a solo miner isn't lucrative financially speaking, but as an independent owner/operator, you are contributing to the decentralization of our network. when you use something like ezminer (looks super sketch, i know) you put the address to your personal wallet instead of an address to a pools wallet.
